### ISSUE TITLE
Feature Request: Better Ref Logix Interaction

### ISSUE THREAD

**Issue by TehTurk** [Frooxius/NeosPublic#360]
Currently  Left Facing "Triangle" Refs in most applications are fine, but would it be possible to have them exist after cutting the wire they connect to? Often A Ref can be a connecting point or a junction for alot of logix, and being able to break them up, or duplicate them would be somewhat handy! After cutting, instead of becoming a Side Triangle, it becomes a square, showing it has no direction, but could point anywhere. This could be also applicable for Right Facing Triangle Refs as well. 

Another is also being able to Generate a Ref point from an output or input. This would allow you to break up Logix a bit more cleanly and also have be able to separately pack certain chains of logix. 

![krita_ZUQmXNwA4O](https://user-images.githubusercontent.com/26357446/72134617-92b6a680-3352-11ea-8acf-a6799d2b3c58.png)

**Comment by shiftyscales** [comment]
In testing this, this issue seems to be invalid. It is possible to cut the wire while retaining the driver as long as the source connection is maintained. You can then connect to a new field by dragging the wire out from the tip of the driver.

**Comment by H3BO3** [comment]
Turk wasn't talking about drive arrows, he was talking about Refs, as stated in his title and issue, explicitly stated as "left-facing".

**Comment by TehTurk** [comment]
Ah so this was a Behaviour Request change with a UI change to show that, Sorry if it was improperly worded or as this was also a bit older and can update wordwise to better reflect it's purpose because you misunderstood I think Shifty. This was suppose to 
A) Allow Refs/Drives to Exist until they are deleted as a Node, vs having them poof once the source is removed by cutting the connection.
B) Duplication of Refs/Drives, currently this doesn't behave in that way unless I haven't tried in awhile.

This was a Behaviour that albeit I was specifically talking Refs as they are Left Facing but other bits also applied to Drives as well if you think about it. Because Left/Right Triangle Nodes are basically References so I didn't think that would get misunderstood.

The UI interaction was more so for Left Facing Ref Behavior, for Drives I realize this is a bit moot as once Target is gone, it's just a relay that needs to be reconnected.

**Comment by shiftyscales** [comment]
Okay, that makes sense I guess.

It appears you already can have multiple ref nodes pointing to the same slots- but it seems there is some pretty buggy behavior associated to it. E.g. I plugged two nodes into the same interface separately, and it formed two reference nodes- but it seemed I was unable to pack both of them at the same time.

Also as has been noted, it's not possible to duplicate the node as it can not exist without a connection. Thanks for clarifying.

**Comment by TehTurk** [comment]
Yes I'm aware, and actually having them not pack together is what I want to keep. Having them be separate packs allows for cleaner separation of LogiX. I do not consider that a bug and that should stay. I repeat, them not packing together isn't a bug, it's because they aren't connected and this has it's inherent benefits in cleanup/organization.

**Comment by shiftyscales** [comment]
You misunderstand, @TehTurk- I literally mean they can't both be packed. Meaning if you try to pack them even separately, the second Logix group won't pack at all.

**Comment by TehTurk** [comment]
Hmm weird can you show me how.you did it? I don't think I've run into this, I know there's a thing currently where you need to hit higher in the chain to pack. If they are connected.
